File: homework01/caesar.py
def encrypt_caesar(plaintext):
    """
    Encrypts plaintext using a Caesar cipher.

    >>> encrypt_caesar("PYTHON")
    'SBWKRQ'
    >>> encrypt_caesar("python")
    'sbwkrq'
    >>> encrypt_caesar("Python3.6")
    'Sbwkrq3.6'
    >>> encrypt_caesar("")
    ''
    """
    ciphertext = ""
    for char in plaintext:
        if "A" <= char <= "z":
            symbol = ord(char) + 3

            if "A" <= char <= "Z" < chr(symbol):
                symbol -= 26

            elif chr(symbol) > "z":
                symbol -= 26

            char = chr(symbol)
        ciphertext += char

    return ciphertext


def decrypt_caesar(ciphertext):
    """
    Decrypts a ciphertext using a Caesar cipher.

    >>> decrypt_caesar("SBWKRQ")
    'PYTHON'
    >>> decrypt_caesar("sbwkrq")
    'python'
    >>> decrypt_caesar("Sbwkrq3.6")
    'Python3.6'
    >>> decrypt_caesar("")
    ''
    """
    plaintext = ""
    for char in ciphertext:
        if "A" <= char <= "z":
            enc = ord(char) - 3

            if enc < ord("A"):
                enc += 26

            if "a" <= char <= "z" and enc < ord("a"):
                enc += 26

            char = chr(enc)
        plaintext += char

    return plaintext


print(encrypt_caesar("Python3.6"))
print(decrypt_caesar("SBWKRQ"))
print(decrypt_caesar("Sbwkrq3.6"))

File: homework01-go/caesar/caesar.go
package caesar

func EncryptCaesar(plaintext string, shift int) string {
	var ciphertext string

	shift = shift % 26
	for _, char := range plaintext {
		if (char >= 'A' && char <= 'Z') || (char >= 'a' && char <= 'z') {
			if (char+rune(shift) > 'Z' && char <= 'Z') || (char+rune(shift) > 'z') {
				char = char - 26
			}
			char = char + rune(shift)
		}
		ciphertext += string(char)
	}

	return ciphertext
}

func DecryptCaesar(ciphertext string, shift int) string {
	var plaintext string

	shift = shift % 26
	for _, char := range ciphertext {
		if (char >= 'A' && char <= 'Z') || (char >= 'a' && char <= 'z') {
			if (char-rune(shift) < 'A' && char >= 'A') || (char-rune(shift) < 'a' && char >= 'a') {
				char = char + 26
			}
			char = char - rune(shift)
		}
		plaintext += string(char)
	}

	return plaintext
}
